src/jupyterlabcontroller/services/lab.py

```diff
@@ -874,7 +878,8 @@ def build_pod_spec(
             restart_policy="OnFailure",
             security_context=V1PodSecurityContext(
                 run_as_non_root=True,
-                fs_group=user.gid,
+                run_as_user=user.uid,
+                run_as_group=user.gid,
```

Comment:
These were previously set in the container rather than the pod level. I'm okay with setting them here instead and overriding them at the container level for init containers, but this needs verification that they can be overridden at the container level when they're set at the pod level (in other words, that this doesn't break privileged init containers).

Also, I'm not sure this does anything. I think the assumption here is that the creation of the pod working directory would be done as the user configured in the pod security context, but is that documented anywhere? It would make more sense to me from a security standpoint if the working directory were created by the Kubernetes control plane, and thus is always created as root regardless of the pod security settings.

Reply:
Container overrides Pod:

```
To specify security settings for a Container, include the securityContext field in the Container manifest. The securityContext field is a [SecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#securitycontext-v1-core) object. Security settings that you specify for a Container apply only to the individual Container, and they override settings made at the Pod level when there is overlap. Container settings do not affect the Pod's Volumes.
```

cf https://kubernetes.io/docs/tasks/configure-pod-container/security-context/

But....I think you're right that it doesn't do anything for us here.

Reply:
Dropping fs_group still seems worthwhile, though.

Reply:
Yeah, no objections to putting this in at the pod level. I'm not sure if it does anything, but it can't hurt.